### PR TITLE
Checks review

### PR DESCRIPTION
**Pull request by shloka-bhalgat-unskript**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

File: AWS/legos/aws_filter_instances_without_termination_and_lifetime_tag/aws_filter_instances_without_termination_and_lifetime_tag.py [hunk 1]
```diff
@@ -33,39 +33,41 @@ def aws_filter_instances_without_termination_and_lifetime_tag_printer(output):
...
-    return instances_dict
+
+def fetch_instances_from_valid_region(reservations, aws_region, termination_tag_name, lifetime_tag_name):
+    result = []
+    right_now = date.today()
+    
+    for reservation in reservations:
+        for instance in reservation.get('Instances', []):
+            instance_id = instance.get('InstanceId')
+            tagged_instance = instance.get('Tags', [])
```
Comment (shloka-bhalgat-unskript) [inline AWS/legos/aws_filter_instances_without_termination_and_lifetime_tag/aws_filter_instances_without_termination_and_lifetime_tag.py:44]: Optimized code to check for Tags

File: AWS/legos/aws_get_iam_users_without_password_policies/aws_get_iam_users_without_password_policies.py [hunk 2]
```diff
@@ -27,14 +27,19 @@ def aws_get_iam_users_without_password_policies(handle) -> Tuple:
...
+        for user in response['Users']:
+            user_name = user['UserName']
+            try:
+                # Check for user-managed policies attached to the user
+                user_policies = iam_client.list_user_policies(UserName=user_name)
+                # Check for AWS-managed policies attached to the user
+                attached_policies = iam_client.list_attached_user_policies(UserName=user_name)
+                # If the user has no policies, add to result
+                if not user_policies['PolicyNames'] and not attached_policies['AttachedPolicies']:
+                    result.append(user_name)
```
[the remark this answers is not in the document]
Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_get_iam_users_without_password_policies/aws_get_iam_users_without_password_policies.py:42]: Yes. Output with the new implementation- 
Screenshot 2023-08-24 at 5 11 48 PM

File: AWS/legos/aws_get_long_running_rds_instances_without_reserved_instances/aws_get_long_running_rds_instances_without_reserved_instances.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ def aws_get_long_running_rds_instances_without_reserved_instances(handle, region
     for reg in all_regions:
         try:
             rdsClient = handle.client('rds', region_name=reg)
-            response = rdsClient.describe_reserved_nodes()
+            response = rdsClient.describe_reserved_db_instances()
```
[the remark this answers is not in the document]
Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_get_long_running_rds_instances_without_reserved_instances/aws_get_long_running_rds_instances_without_reserved_instances.py:47]: Yes the API was incorrect. The correct one is `describe_reserved_db_instances() `. Updated output for this check lego- 
Screenshot 2023-08-24 at 5 56 51 PM

File: Kubernetes/legos/k8s_get_all_evicted_pods_from_namespace/k8s_get_all_evicted_pods_from_namespace.py [hunk 4]
```diff
@@ -39,33 +39,37 @@ def k8s_get_all_evicted_pods_from_namespace(handle, namespace: str = "") -> Tupl
...
-    if len(result) != 0:
+        for pod in pod_details.get('items', []):
+            if pod['status']['phase'] == 'Failed' and any(cs.get('reason') == 'Evicted' for cs in pod['status']['conditions']):
+                pod_dict = {
+                    "pod_name": pod["metadata"]["name"],
+                    "namespace": pod["metadata"]["namespace"]
+                }
+                result.append(pod_dict)
+    except Exception as e:
+        raise e
```
[the remark this answers is not in the document]
Comment (shloka-bhalgat-unskript) [reply Kubernetes/legos/k8s_get_all_evicted_pods_from_namespace/k8s_get_all_evicted_pods_from_namespace.py:71]: It was broken earlier. The stats should be "failed" and reason should be "evicted". Earlier we were grep'ing for "Evicted" pods which was incorrect.

File: AWS/legos/aws_get_schedule_to_retire_instances/aws_get_schedule_to_retire_instances.py [hunk 5]
```diff
@@ -34,35 +34,29 @@ def aws_get_schedule_to_retire_instances( handle, region: str=None) -> Tuple:
...
-            output = aws_filter_ec2_instances(handle=handle,region=r)
-            if len(output)!=0:
+            output = aws_filter_ec2_instances(handle=handle, region=r)
+            if output:
                 for o in output:
-                    all_instances_dict = {}
-                    all_instances_dict["region"]=r
-                    all_instances_dict["instance"]=o
-                    all_instances.append(all_instances_dict)
+                    all_instances_dict = {"region": r, "instance": o}
```
[the remark this answers is not in the document]
Comment (shloka-bhalgat-unskript) [reply AWS/legos/aws_get_schedule_to_retire_instances/aws_get_schedule_to_retire_instances.py:45]: That's correct. I've fixed the loop to record it only once.

**Review by abhishek-unskript — APPROVED** [review]

**Pull request merged**